include/nbl/asset/filters/CBlitUtilities.h

```diff
@@ -129,19 +158,58 @@ class CBlitUtilities : public IBlitUtilities
 
 				float relativePos = outPixelCenter - float(windowCoord); // relative position of the last pixel in window from current (ith) output pixel having a unique phase sequence of kernel evaluation points
 
+				double accum[ChannelCount] = { };
 				for (int32_t j = 0; j < _windowSize; ++j)
 				{
-					for (uint32_t ch = 0; ch < MaxChannels; ++ch)
+					for (uint32_t ch = 0; ch < ChannelCount; ++ch)
 					{
 						const double weight = static_cast<double>(kernel.weight(relativePos, ch));
-						if constexpr (std::is_same_v<LutDataType, uint16_t>)
-							outKernelWeightsPixel[(i * _windowSize + j) * MaxChannels + ch] = core::Float16Compressor::compress(float(weight));
+						if (!shouldNormalize)
+						{
+							if constexpr (std::is_same_v<LutDataType, uint16_t>)
+								outKernelWeightsPixel[(i * _windowSize + j) * ChannelCount + ch] = core::Float16Compressor::compress(float(weight));
+							else
+								outKernelWeightsPixel[(i * _windowSize + j) * ChannelCount + ch] = LutDataType(weight);
+						}
 						else
-							outKernelWeightsPixel[(i * _windowSize + j) * MaxChannels + ch] = LutDataType(weight);
+						{
+							accum[ch] += weight;
+							outKernelWeightsPixel_f64[(i * _windowSize + j) * ChannelCount + ch] = weight;
+						}
 					}
 					
 					relativePos -= 1.f;
 				}
+
+				if (shouldNormalize)
+				{
+					double normalizationFactor[ChannelCount] = { };
+					std::fill(normalizationFactor, normalizationFactor + ChannelCount, 1.0);
```

Comment:
there's a better way, if all weights are close to 0 and their sum was supposed to equal `normalizeWeightsTo`

if you detect `accum[ch]<threshold` then fill all weights with `normalizeWeightsTo/double(_windowSize)`

Reply:
I'm leaving it to @achalpandeyy

Reply:
Resolved.